trace-normalization/src/normalizer.rs

```diff
@@ -103,11 +113,71 @@ pub(crate) fn is_valid_status_code(sc: &str) -> bool {
     false
 }
 
+// normalize_trace takes a trace and
+// * returns an error if there is a trace ID discrepancy between 2 spans
+// * returns an error if at least one span cannot be normalized
+pub fn normalize_trace(trace: &mut [pb::Span]) -> anyhow::Result<()> {
+    if trace.is_empty() {
+        anyhow::bail!("Normalize Trace Error: Trace is empty.");
+    }
+
+    let first_trace_id = trace[0].trace_id;
```

Comment:
I think it might be safer to use get(0) which will return a result, then you can check the error.
having decorrelated `is_empty` and `[0]` split could be a risk later?

Reply:
good idea, added